langchain/document_transformers/text_extract.py

```diff
@@ -0,0 +1,42 @@
+from typing import Any, Sequence
+
+from langchain.schema import BaseDocumentTransformer, Document
+from langchain.utils import get_from_env
+
+
+class DoctranPropertyExtractor(BaseDocumentTransformer):
+    """Extracts properties from text documents using doctran."""
+
+    def transform_documents(
+        self, documents: Sequence[Document], **kwargs: Any
+    ) -> Sequence[Document]:
+        raise NotImplementedError
+
+    async def atransform_documents(
+        self, documents: Sequence[Document], **kwargs: Any
+    ) -> Sequence[Document]:
+        """Extracts properties from text documents using doctran."""
+
+        properties = kwargs.get("properties", None)
```

Comment:
should this and openai_api_key just be passed in during init?

Reply:
threw lint error from `@abstractmethod`:

```
    @abstractmethod
    async def atransform_documents(
        self, documents: Sequence[Document], **kwargs: Any
    ) -> Sequence[Document]:
```

Reply:
as in to the class `__init__` method, and accessed as instance attributes in the function

Reply:
ya that's reasonable; none of these classes were using init methods. 

@jasonwcfan will take care of this.